src/pages/index.js

```diff
@@ -0,0 +1,302 @@
+const avatarInput = document.querySelector("#avatar-input");
+
+//delete card Modal
+const deleteCardModal = document.querySelector("#delete-card-modal");
+const deleteForm = document.querySelector("#delete-card-form");
+const deleteExitBtn = document.querySelector(".modal__delete-exit-btn");
+const deleteCancel = document.querySelector("#delete-card-cancel");
+const deleteSubmitBtn = document.querySelector("#delete-submit-button");
+
+let selectedCard;
+let selectedCardId;
+
+newModalCloseButton.addEventListener("click", () => {
+  closeModal(newPostModal);
+});
+
+profileAddButton.addEventListener("click", () => {
+  openModal(newPostModal);
+});
+
+profileExitButton.addEventListener("click", () => {
+  closeModal(profileModal);
+});
+
+avatarModalBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarCloseBtn.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+deleteExitBtn.addEventListener("click", () => {
+  closeModal(deleteCardModal);
+});
+
+function handleAvatarSubmit(evt) {
+  // form for Avatar changing
+  evt.preventDefault();
+  setButtonText(avatarSubmitButton, true);
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      avatarImg.src = data.avatar;
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarSubmitButton, false);
+    });
+  closeModal(avatarModal);
+}
+
+function handleImageFormSubmit(evt) {
+  //new post modal submit
+  evt.preventDefault();
+  setButtonText(cardSubmitButton, true);
+
+  const name = caption.value;
+  const link = imageLink.value;
+  api
+    .createNewCard(name, link)
+    .then((card) => {
+      const cardsElement = getCardElement(card);
+
+      cardsList.prepend(cardsElement);
+      evt.target.reset();
+      closeModal(newPostModal);
+      resetValidation(evt.target, validationConfig);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitButton, false);
+    });
+}
+
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
+modalImageForm.addEventListener("submit", handleImageFormSubmit);
+
+profileEditButton.addEventListener("click", () => {
+  profileName.value = nameOnPage.textContent;
+  profileDescription.value = descriptionOnPage.textContent;
+
+  openModal(profileModal);
+  resetValidation(profileModal, validationConfig);
+});
+
+function handleProfileFormSubmit(evt) {
+  //form for profile name and description
+  evt.preventDefault();
+  //setButtonText(profileSubmitBtn, true);
+  profileSubmitBtn.textContent = "saving...";
+  api
+    .editUserInfo({ name: profileDescription.value, about: profileName.value })
+    .then((data) => {
+      nameOnPage.textContent = data.about;
+      descriptionOnPage.textContent = data.name;
+      closeModal(profileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      //setButtonText(profileSubmitBtn, false);
+      profileSubmitBtn.textContent = "save";
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions